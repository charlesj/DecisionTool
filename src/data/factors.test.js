import { fromJS } from 'immutable'
import shortid from 'shortid'
import {
  addFactor,
  updateFactorName,
  updateFactorWeight,
  removeFactor,
} from './factors'

jest.mock('shortid')

const initState = fromJS({})

let id = 0

beforeEach(() => {
  shortid.mockImplementation(() => `id-${id++}`)
})

describe('addFactor', () => {

  it('adds the factor to the state', () => {
    expect(addFactor(initState)).toMatchSnapshot()
  })
})

describe('updateFactorName', () => {
  it('updates the name of the factor given', () => {
    const factorId = 'factor-id-1'
    const state = fromJS({
      factors: {
        [factorId]: { name: 'old name' }
      }
    })

    expect(updateFactorName(state, factorId, 'new name')).toMatchSnapshot()
  })
})

describe('updateFactorWeight', () => {
  it('updates the weight of the factor', () => {
    const factorId = 'factor-id-1'
    const state = fromJS({
      factors: {
        [factorId]: { name: 'old name' }
      }
    })

    expect(updateFactorWeight(state, factorId, '3')).toMatchSnapshot()
  })
})

describe('removeFactor', () => {
  it('removes the given factor', () => {
    const factorId = 'factor-id-1'
    const state = fromJS({
      factors: {
        [factorId]: { name: 'old name' },
        factor2: { name: 'this factor stays' }
      }
    })
    expect(removeFactor(state, factorId)).toMatchSnapshot()
  })
})
