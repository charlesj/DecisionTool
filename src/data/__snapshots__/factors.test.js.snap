// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addFactor adds the factor to the state 1`] = `
Immutable.Map {
  "factors": Immutable.Map {
    "id-0": Immutable.Map {
      "name": "factor id-0",
    },
  },
}
`;

exports[`removeFactor removes the given factor 1`] = `
Immutable.Map {
  "factors": Immutable.Map {
    "factor2": Immutable.Map {
      "name": "this factor stays",
    },
  },
}
`;

exports[`updateFactorName updates the name of the factor given 1`] = `
Immutable.Map {
  "factors": Immutable.Map {
    "factor-id-1": Immutable.Map {
      "name": "new name",
    },
  },
}
`;

exports[`updateFactorWeight updates the weight of the factor 1`] = `
Immutable.Map {
  "factors": Immutable.Map {
    "factor-id-1": Immutable.Map {
      "name": "old name",
      "weight": "3",
    },
  },
}
`;
